datafusion/core/src/physical_plan/metrics/tracker.rs

```diff
@@ -1,104 +0,0 @@
-// Licensed to the Apache Software Foundation (ASF) under one
-// or more contributor license agreements.  See the NOTICE file
-// distributed with this work for additional information
-// regarding copyright ownership.  The ASF licenses this file
-// to you under the Apache License, Version 2.0 (the
-// "License"); you may not use this file except in compliance
-// with the License.  You may obtain a copy of the License at
-//
-//   http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing,
-// software distributed under the License is distributed on an
-// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-// KIND, either express or implied.  See the License for the
-// specific language governing permissions and limitations
-// under the License.
-
-//! Metrics with memory usage tracking capability
-
-use crate::physical_plan::metrics::{
-    BaselineMetrics, Count, ExecutionPlanMetricsSet, Time,
-};
-use std::sync::Arc;
-use std::task::Poll;
-
-use crate::error::Result;
-use crate::execution::memory_pool::{MemoryConsumer, MemoryPool, MemoryReservation};
-use arrow::record_batch::RecordBatch;
-
-/// Wraps a [`BaselineMetrics`] and records memory usage on a [`MemoryReservation`]
-#[derive(Debug)]
-pub struct MemTrackingMetrics {
```

Comment:
I think this construction may date from an earlier iteration of the memory tracking, as it stands now it makes little sense.

The memory reported by the mem_used metric will be a somewhat arbitrary value based on the last value at the point the plan finished. Additionally there isn't any way to use the MemoryReservation in a fallible manner. It felt easier to just separate the concerns of reporting plan metrics from tracking runtime memory usage.